File: model/src/main/java/com/itransition/fanfictionbackend/model/Rating.java
package com.itransition.fanfictionbackend.model;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@Entity
@Table(name = "rating")
@EqualsAndHashCode(callSuper = true)
public class Rating extends WithLongId {

    private Float value;

    @ManyToOne(optional = false)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(optional = false)
    @JoinColumn(name = "fanfic_id")
    private Fanfic fanfic;
}

File: src/main/kotlin/com/itransition/fanfictionbackend/model/Rating.kt
package com.itransition.fanfictionbackend.model

import javax.persistence.Entity
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.Table

@Entity
@Table(name = "rating")
class Rating : WithLongId() {

    var value: Float? = null

    @ManyToOne(optional = false)
    @JoinColumn(name = "user_id")
    var user: User? = null

    @ManyToOne(optional = false)
    @JoinColumn(name = "fanfic_id")
    var fanfic: Fanfic? = null
}
